.github/workflows/relativity_main.yml

```diff
@@ -1,36 +1,105 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI
 
-# Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the relativity-main branch
   push:
     branches: [ relativity-main ]
   pull_request:
     branches: [ relativity-main ]
 
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
-
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
+  build-spark-operator:
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+
+      - name: Run gofmt check
+        run: make fmt-check
+
+      - name: Run static analysis
+        run: make static-analysis
+
+      - name: Run unit tests
+        run: make unit-test
+
+      - name: Azure login
+        uses: Azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
+      - name: Get secrets
+        uses: Azure/get-keyvault-secrets@v1
+        with:
+          keyvault: ${{ secrets.KEY_VAULT }}
+          secrets: "acrUsername, acrPassword"
+        id: kv
+
+      - name: Login to docker
+        uses: docker/login-action@v1 
+        with:
+          registry: ${{ secrets.REGISTRY }}
+          username: ${{ steps.kv.outputs.acrUsername }}
+          password: ${{ steps.kv.outputs.acrPassword }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build image
+        uses: docker/build-push-action@v2
+        with:
+          tags: spark-operator:local
+          outputs: 'type=docker,dest=/tmp/base-image.tar'
+          build-args: |
+            SPARK_IMAGE=${{ secrets.BASE_IMAGE }}
+
+      - name: Cache image
```

Comment:
I used it as a functional crutch vs performance, where I'm not publishing the image to the repository but need some way to reference it from a different job. I have the prisma scan in a different job because I couldn't find a great way for it not to fail the job and wanted to split the indicators for a failure to build the image and scan failure